src/com/lftechnology/java/training/niraj/ems/services/implementations/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+        Employee employee = new Employee();
+        Map<String, String> condition = new LinkedHashMap<String, String>();
+        condition.put(employee.getPrimaryKey(), s);
+        List<Map<String, String>> resultList = findQuery(conn, employee, condition, Operators.AND);
+        parseEmployeeObject(resultList, employee);
+        return employee;
+    }
+
+    @Override
+    public Employee findByAttributes(Map<String, String> conditions) throws SQLException {
+        Employee employee = new Employee();
+        List<Map<String, String>> resultList = findQuery(conn, employee, conditions, Operators.AND);
+        parseEmployeeObject(resultList, employee);
+        return employee;
+    }
+
+    @Override
+    public boolean update(Employee t) throws SQLException, CustomException {
+        if (t.getId() != null) {
+            return updateQuery(conn, t) > 0 ? true : false;
+        } else {
+            throw new CustomException("Invalid Employee Object");
+        }
+    }
+
+    @Override
+    public List<Employee> findAll() throws SQLException {
+        Employee employee = new Employee();
+        Employee searchResult;
+        List<Employee> employeeList = new ArrayList<Employee>();
+        List<Map<String, String>> results = findQuery(conn, employee);
+        System.out.println(results);
+        for (Map<String, String> result : results) {
+            searchResult = new Employee();
+            setEmployeeObject(result, searchResult);
+            employeeList.add(searchResult);
+        }
+        return employeeList;
+
+    }
+
+    @Override
+    public List<Employee> findAll(Map<String, String> conditions) throws SQLException {
+        Employee employee = new Employee();
+        Employee searchResult;
+        List<Employee> employeeList = new ArrayList<Employee>();
+        List<Map<String, String>> results = findQuery(conn, employee, conditions, Operators.AND);
+        for (Map<String, String> result : results) {
+            searchResult = new Employee();
+            setEmployeeObject(result, searchResult);
+            employeeList.add(searchResult);
+        }
+        return employeeList;
+    }
+
+    @Override
+    public List<Employee> findAll(Map<String, String> conditions, Operators operator) throws SQLException {
+        Employee employee = new Employee();
+        Employee searchResult;
+        List<Employee> employeeList = new ArrayList<Employee>();
+        List<Map<String, String>> results = findQuery(conn, employee, conditions, operator);
+        for (Map<String, String> result : results) {
+            searchResult = new Employee();
+            setEmployeeObject(result, searchResult);
+            employeeList.add(searchResult);
+        }
+        return employeeList;
+    }
+
+    @Override
+    public boolean exists(Map<String, String> conditions) throws SQLException {
+        int count = getCount(conditions);
+        System.out.println(count);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)